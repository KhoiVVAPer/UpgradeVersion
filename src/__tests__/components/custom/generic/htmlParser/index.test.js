import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import HtmlParser from '../../../../../components/custom/generic/htmlParser';

const setUp = (props) => {
  const component = shallow(<HtmlParser {...props} />);
  return component;
};

describe('components > custom > elements > HtmlParser', () => {
  let component;
  const props = {
    html: '<p>hello</p>'
  };
  beforeEach(() => {
    component = setUp(props);
    return component;
  });

  it('Check snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders component', () => {
    const input = component.find('View').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders HTML component', () => {
    const input = component.find('HTML').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders HTML component html prop', () => {
    const input = component.find('HTML').prop('html');
    expect(input).toBeDefined();
  });
});
