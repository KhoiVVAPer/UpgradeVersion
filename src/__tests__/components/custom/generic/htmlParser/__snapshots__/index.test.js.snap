// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > custom > elements > HtmlParser Check snapshot 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          hello
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;
